### PR TITLE
Allow colon in image names

### PR DESCRIPTION
**Pull request by DrDaveD**
**Description of the Pull Request (PR):**

This PR allows colons to be in image names without being in transport:ref format.  Without this, if an image name or sandbox path has a colon, singularity dies with an error message "Unsupported transport type".  The PR also moves the "IsValidURI" function from a place where it makes little sense to be, build/conveyerPacker.go, to util/uri/uri.go.

**This fixes or addresses the following GitHub issues:**

None

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [user](https://www.github.com/sylabs/singularity-userdocs) and [administrator](https://www.github.com/sylabs/singularity-admindocs) documentation bases.
- [x] I have tested this PR locally with a `make testall`
- [x] This PR is against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularity-maintainers

### CODE REVIEW

File: src/pkg/build/build.go [hunk 1]
```diff
@@ -357,7 +358,7 @@ func getcp(def types.Definition, libraryURL, authToken string) (ConveyorPacker,
 
 // makeDef gets a definition object from a spec
 func makeDef(spec string) (types.Definition, error) {
-	if ok, err := IsValidURI(spec); ok && err == nil {
+	if ok, err := uri.IsValidURI(spec); ok && err == nil {
```
Comment (ArangoGutierrez) [inline src/pkg/build/build.go:361]: If you are going to propose a `uri` pkg, `IsValidURI()` should be renamed `IsValid()`

**Comment by jscook2345** [comment]
Looking at your changes I can see the reasoning behind this request. After reviewing it I think it looks OK.

Note, the colon character has to be escaped in the shell when used and thus leads to a few issues with usage. It also is an illegal character if you're sharing files with a Windows-based system.

**Review by ArangoGutierrez — DISMISSED** [review]

File: src/pkg/build/build.go (hunk 1 above)
Comment (DrDaveD) [reply src/pkg/build/build.go:361]: @ArangoGutierrez There already was a uri package with SplitURI and NameFromURI.  Must those change to drop the URI too?  I was just following the conventions I saw.

**Comment by DrDaveD** [comment]
@jscook2345 Colons do not need to be escaped from any Linux shells that I am familiar with.

File: src/pkg/build/build.go (hunk 1 above)
Comment (ArangoGutierrez) [reply src/pkg/build/build.go:361]: I will believe so, personally I don't like `pkg.FuncPkg` redundancy , I prefer `pkg.Func` .
But @bauerm97 is the one to call func names.

Comment (DrDaveD) [reply src/pkg/build/build.go:361]: In particular what would NameFromURI change to?  uri.NameFrom?   That's pretty awkward.

Comment (ArangoGutierrez) [reply src/pkg/build/build.go:361]: we accept suggestions ;P 
I guess `uri.GetName()` sounds nice.....

Comment (DrDaveD) [reply src/pkg/build/build.go:361]: Ok that sounds reasonable.  That's the name I thought of after I posted too.  

I think this PR is big enough however.  I suggest that we merge this as-is and then I'll submit another PR with the uri package name changes.

Comment (ArangoGutierrez) [reply src/pkg/build/build.go:361]: that sounds reasonable.
I'll leave @bauerm97 proceed with this PR

**Comment by DrDaveD** [comment]
I rebased on the current master.  @bauerm97, can we please proceed with this PR?

File: src/pkg/build/build.go (hunk 1 above)
Comment (yhcote) [reply src/pkg/build/build.go:361]: I am inclined to merge this one. @DrDaveD : Do you have a list of names you wanna refactor in this package?

Comment (DrDaveD) [reply src/pkg/build/build.go:361]: I'll rename all of them in a separate PR: NameFromURI, SplitURI, and IsValidURI

Comment (yhcote) [reply src/pkg/build/build.go:361]: Ok Sounds good

**Review by yhcote — APPROVED** [review]
Looks good to me.

**Pull request merged**